src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1927,17 +1928,26 @@ internal ImmutableArray<Diagnostic> GetDiagnostics(CompilationStage stage, bool
                             var syntaxTree = syntaxTrees[i];
                             AppendLoadDirectiveDiagnostics(builder, _syntaxAndDeclarations, syntaxTree);
                             builder.AddRange(syntaxTree.GetDiagnostics(cancellationToken));
+                            if (parseOptionsReported.Add(syntaxTree.Options))
+                            {
+                                builder.AddRange(syntaxTree.Options.Errors);
```

Comment:
>builder.AddRange(syntaxTree.Options.Errors) [](start = 32, length = 43)

What is going to be the location for the errors? How a client can figure out what parse options are causing them? I think the location of the errors should be adjusted to point to the associated syntax tree.